Configs: Macros: Changes in the macro statements

LipinskiPNordicSemi opened the pull request:
Changes in the macro statements that allows to build with -Wudef flag enebaled, without errors.
```

/home/patl/ncs/v2.5.0/zephyr/modules/hal_nordic/nrfx/./nrfx_glue.h:360:5: error: "NRF_802154_VERIFY_PERIPHS_ALLOC_AGAINST_MPSL" is not defined, evaluates to 0 [-Werror=undef]
  360 | #if NRF_802154_VERIFY_PERIPHS_ALLOC_AGAINST_MPSL
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/patl/ncs/v2.5.0/zephyr/include/zephyr/logging/log_core.h:9,
                 from /home/patl/ncs/v2.5.0/zephyr/include/zephyr/logging/log.h:11,
                 from /home/patl/ncs/v2.5.0/zephyr/include/zephyr/net/net_core.h:19,
                 from /home/patl/ncs/v2.5.0/zephyr/include/zephyr/net/net_if.h:25,
                 from /home/patl/ncs/v2.5.0/zephyr/include/zephyr/net/net_context.h:28:
/home/patl/ncs/v2.5.0/zephyr/include/zephyr/logging/log_msg.h:82:5: error: "CONFIG_LOG_THREAD_ID_PREFIX" is not defined, evaluates to 0 [-Werror=undef]
   82 | #if CONFIG_LOG_THREAD_ID_PREFIX
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/patl/ncs/v2.5.0/zephyr/include/zephyr/logging/log_msg.h:669:5: error: "CONFIG_LOG_THREAD_ID_PREFIX" is not defined, evaluates to 0 [-Werror=undef]
  669 | #if CONFIG_LOG_THREAD_ID_PREFIX
      |     ^~~~~~~~~~~~~~~~~~~~~~~~~~~
In file included from /home/patl/ncs/v2.5.0/zephyr/include/zephyr/fs/fs.h:14,
                 from /home/patl/ncs/v2.5.0/zephyr/include/zephyr/sys/fdtable.h:12,
                 from /home/patl/ncs/v2.5.0/zephyr/include/zephyr/net/socket.h:31:
/home/patl/ncs/v2.5.0/zephyr/include/zephyr/fs/fs_interface.h:16:6: error: "CONFIG_FILE_SYSTEM_MAX_FILE_NAME" is not defined, evaluates to 0 [-Werror=undef]
   16 | #if (CONFIG_FILE_SYSTEM_MAX_FILE_NAME - 0) > 0
      |      ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
cc1plus: all warnings being treated as errors
ninja: build stopped: subcommand failed.
```

LuDuda reviewed (approved):
Thx!

de-nordic reviewed (changes requested):
You need to change the commit subject, this one will not work.
I am not even sure this can be covered by "treewide" commit, probably will have to go as three different commits, one for each subsystem/module.

LipinskiPNordicSemi committed:
fs: Changes in the macro statements

Changes in the macro statements that allows to build
with -Wudef flag enebaled, without errors.

Signed-off-by: Patryk Lipinski <patryk.lipinski@nordicsemi.no>

LipinskiPNordicSemi committed:
logging: Changes in the macro statements

Changes in the macro statements that allows to build
with -Wudef flag enebaled, without errors.

Signed-off-by: Patryk Lipinski <patryk.lipinski@nordicsemi.no>

LipinskiPNordicSemi committed:
modules: hal_nordic: nrfx_glue: Changes in the macro statements

Changes in the macro statements that allows to build
with -Wudef flag enebaled, without errors.

Signed-off-by: Patryk Lipinski <patryk.lipinski@nordicsemi.no>
